src/pages/projectSettings/tabs/EventLogTab/useEvents.ts

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+import { useQuery } from "@apollo/client";
+import { useToastContext } from "context/toast";
+import {
+  ProjectEventLogsQuery,
+  ProjectEventLogsQueryVariables,
+  RepoEventLogsQuery,
+  RepoEventLogsQueryVariables,
+} from "gql/generated/types";
+import { GET_PROJECT_EVENT_LOGS, GET_REPO_EVENT_LOGS } from "gql/queries";
+
+export const useEvents = (
+  identifier: string,
+  isRepo: boolean,
+  limit: number
+) => {
+  const dispatchToast = useToastContext();
+
+  const [prevCount, setPrevCount] = useState(0);
+  const [showLoadButton, setShowLoadButton] = useState(true);
+
+  // Hide Load More button when event count < limit is returned,
+  // or when the same event count is returned twice in a row.
+  const onCompleted = (eventData) => {
+    if (eventData.length === prevCount) {
+      setShowLoadButton(false);
+    } else if (eventData.length - prevCount < limit) {
+      setShowLoadButton(false);
+    } else {
+      setPrevCount(eventData.length);
+    }
+  };
+
+  const { data: projectEventData, fetchMore: projectFetchMore } = useQuery<
+    ProjectEventLogsQuery,
+    ProjectEventLogsQueryVariables
+  >(GET_PROJECT_EVENT_LOGS, {
+    variables: { identifier, limit },
+    errorPolicy: "all",
```

Comment:
I see that this was here before but could you remind me why we used this errorPolicy here?

Reply:
Because of type changes over time and the fact that previous events don't necessarily adhere to our current types, we found that we were encountering errors somewhat often when trying to load event logs. But type safety isn't super important for a log like this; it's better to show a log then omit it, so we don't want to block on errors. (Added in https://github.com/evergreen-ci/spruce/pull/1283#issue-1243343462).